[2024-09-20T13:31:07.936+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: weekly_aggregate.spark_weekly_aggregate manual__2024-09-20T13:31:04.349843+00:00 [queued]>
[2024-09-20T13:31:08.004+0000] {taskinstance.py:1083} INFO - Dependencies all met for <TaskInstance: weekly_aggregate.spark_weekly_aggregate manual__2024-09-20T13:31:04.349843+00:00 [queued]>
[2024-09-20T13:31:08.006+0000] {taskinstance.py:1279} INFO - 
--------------------------------------------------------------------------------
[2024-09-20T13:31:08.007+0000] {taskinstance.py:1280} INFO - Starting attempt 1 of 2
[2024-09-20T13:31:08.009+0000] {taskinstance.py:1281} INFO - 
--------------------------------------------------------------------------------
[2024-09-20T13:31:08.094+0000] {taskinstance.py:1300} INFO - Executing <Task(SparkSubmitOperator): spark_weekly_aggregate> on 2024-09-20 13:31:04.349843+00:00
[2024-09-20T13:31:08.116+0000] {standard_task_runner.py:55} INFO - Started process 215 to run task
[2024-09-20T13:31:08.134+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'weekly_aggregate', 'spark_weekly_aggregate', 'manual__2024-09-20T13:31:04.349843+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/weekly_aggregate_dag.py', '--cfg-path', '/tmp/tmp6g1_r0o0']
[2024-09-20T13:31:08.148+0000] {standard_task_runner.py:83} INFO - Job 2: Subtask spark_weekly_aggregate
[2024-09-20T13:31:08.253+0000] {logging_mixin.py:137} WARNING - /home/***/.local/lib/python3.7/site-packages/***/settings.py:249 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-09-20T13:31:08.549+0000] {task_command.py:388} INFO - Running <TaskInstance: weekly_aggregate.spark_weekly_aggregate manual__2024-09-20T13:31:04.349843+00:00 [running]> on host 6320e57075aa
[2024-09-20T13:31:09.145+0000] {taskinstance.py:1509} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=weekly_aggregate
AIRFLOW_CTX_TASK_ID=spark_weekly_aggregate
AIRFLOW_CTX_EXECUTION_DATE=2024-09-20T13:31:04.349843+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2024-09-20T13:31:04.349843+00:00
[2024-09-20T13:31:09.205+0000] {base.py:73} INFO - Using connection ID 'spark_default' for task execution.
[2024-09-20T13:31:09.329+0000] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master yarn --name spark_weekly_aggregate --verbose --queue root.default /opt/***/dags/spark_aggregate.py 2024-09-20
[2024-09-20T13:31:24.930+0000] {spark_submit.py:490} INFO - Using properties file: null
[2024-09-20T13:31:25.937+0000] {spark_submit.py:490} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2024-09-20T13:31:25.940+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:650)
[2024-09-20T13:31:25.941+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:281)
[2024-09-20T13:31:25.944+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:237)
[2024-09-20T13:31:25.946+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:122)
[2024-09-20T13:31:25.948+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1094)
[2024-09-20T13:31:25.949+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1094)
[2024-09-20T13:31:25.951+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
[2024-09-20T13:31:25.953+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1111)
[2024-09-20T13:31:25.956+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1120)
[2024-09-20T13:31:25.957+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2024-09-20T13:31:26.060+0000] {taskinstance.py:1768} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name spark_weekly_aggregate --verbose --queue root.default /opt/***/dags/spark_aggregate.py 2024-09-20. Error code is: 1.
[2024-09-20T13:31:26.081+0000] {taskinstance.py:1323} INFO - Marking task as UP_FOR_RETRY. dag_id=weekly_aggregate, task_id=spark_weekly_aggregate, execution_date=20240920T133104, start_date=20240920T133107, end_date=20240920T133126
[2024-09-20T13:31:26.166+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 2 for task spark_weekly_aggregate (Cannot execute: spark-submit --master yarn --name spark_weekly_aggregate --verbose --queue root.default /opt/***/dags/spark_aggregate.py 2024-09-20. Error code is: 1.; 215)
[2024-09-20T13:31:26.236+0000] {local_task_job.py:208} INFO - Task exited with return code 1
[2024-09-20T13:31:26.322+0000] {taskinstance.py:2578} INFO - 0 downstream tasks scheduled from follow-on schedule check
